File: Programmers/Lv. 2/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    
    def dfs(idx, result):
        if idx == len(numbers):
            if result == target:
                nonlocal answer
                answer += 1                
                return
        else:
            dfs(idx+1, result+numbers[idx])
            dfs(idx+1, result-numbers[idx])
        
    dfs(0,0)
    
    return answer


from collections import deque

def solution(numbers, target):
    answer = 0
    
    q = deque()
    q.append([numbers[0], 0])
    q.append([-numbers[0], 0])
    while q:
        num, cnt = q.popleft()
        if cnt == len(numbers) - 1:
            if num == target:
                answer += 1
        else:
            cnt += 1
            q.append([num + numbers[cnt], cnt])
            q.append([num - numbers[cnt], cnt])
    
    
    return answer

File: Programmers/Lv. 2/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

int answer = 0;
void makeExpression(vector<int> v, int cur_value, int depth, int r, int t)
{
    if (depth == r)
    {
        if (cur_value == t)
        {
            answer++;    
        }        
        return;
    }
    
    int new_value = cur_value + v[depth];
    makeExpression(v, new_value, depth + 1, r, t);

    new_value = cur_value - v[depth];
    makeExpression(v, new_value, depth + 1, r, t);
}
int solution(vector<int> numbers, int target) {
        
    makeExpression(numbers, 0, 0, numbers.size(), target);
    
    return answer;
}